middle_end/flambda2/types/env/cached_level.ml

```diff
@@ -147,15 +147,24 @@ let merge t1 t2 =
   in
   { names_to_types; aliases; symbol_projections }
 
-let remove_unused_closure_vars { names_to_types; aliases; symbol_projections }
-    ~used_closure_vars =
+let remove_unused_closure_vars_and_shortcut_aliases
+    { names_to_types; aliases; symbol_projections } ~used_closure_vars =
+  let canonicalise simple =
```

Comment:
I'm not sure what we get by returning `canonicalise` separately. It's just a closure that contains `aliases`, no? Why not just define `Cached_level.canonicalise` (and `Typing_env.canonicalise` delegating to it)?

Reply:
Maybe I assumed at some point that it would be a map, that would only need to be constructed once... I'm going to implement your suggestion, it is indeed simpler.

Reply:
I've implemented the suggestion to make `canonicalise` a separate function. But I've kept the return type for `Typing_env.Pre_serializable.create`, as I think it makes sense for this function to specify that the canonicalisation function must be applied to all types that live outside the typing environment (such as return types).

Reply:
That makes sense, especially if we add a comment to `create` with that specification